torch/nested/_internal/ops.py

```diff
@@ -380,8 +382,31 @@ def prod_dim_int(func, *args, **kwargs):
     return NestedTensor(func(inp._values, **new_kwargs), **extract_kwargs(args[0]))
 
 
+# Ordinarily, if a program calls .tolist compiling still works because there is
+# special handling in dynamo, but for tensor subclasses if .tolist is called
+# inside torch dispatch, the .tolist call will be directly on a FakeTensor.
+# This would result in an error since wrapper subclasses don't have storage.
+# To avoid this, we handle the fake tensor case by (1) specializing on the size
+# of the tensor to create the output Python list, and (2) creating unbacked
+# symints for each element of the list.
+def sym_tolist(x):
+    if is_fake(x):
+        assert x.dim() == 1 and is_symbolic(x.shape[0])
+        shape_env = x.shape[0].node.shape_env
+        out = []
+        # Specialize on the length of the list
+        for _ in range(x.shape[0]):
+            s = shape_env.create_unbacked_symint()
+            # max value?
+            torch._constrain_as_size(s, min=2)
+            out.append(s)
+        return out
+    return x.cpu().tolist()
```

Comment:
does this logic not exist already somewhere?

Reply:
Hmm I wasn't sure that it did because calling tolist in python below the torch IR wasn't really possible before  torch dispatch subclass support was added to torch compile.